web/src/main/webapp/WEB-INF/config-security/config-security-core.xml

```diff
@@ -143,8 +143,20 @@
     <property name="allowSessionCreation" value="true"/>
   </bean>
 
-  <bean id="formLoginFilter"
-        class="org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter">
+  <bean class="org.springframework.security.web.util.matcher.AntPathRequestMatcher" id="catalogMatcher">
+    <constructor-arg name="pattern" value="/*/*/catalog.*/**"/>
+  </bean>
+
+  <bean class="org.springframework.security.web.util.matcher.AntPathRequestMatcher" id="adminMatcher">
```

Comment:
Not sure if additional urls are required, but instead of using different matchers (unless for any reason each matcher would require some special treatment), would not be better to configure `GeoNetworkHttpSessionRequestCache` with a set of patterns, like in https://github.com/geonetwork/core-geonetwork/blob/develop/web/src/main/webapp/WEB-INF/config-security/config-security-core.xml#L367-L373 ?

Reply:
AntPathRequestMatcher supports a single pattern as constructor, I don't know if it's possible to cover both urls patterns with a single regular expression, because it looks a bit limited https://docs.spring.io/spring/docs/3.2.x/javadoc-api/org/springframework/util/AntPathMatcher.html
You're right, I will try with a RegexRequestMatcher or something similar, I don't think there are reason to give different treatments to different urls.

Reply:
Ok, but `GeoNetworkHttpSessionRequestCache` is a custom class, I think you can send a `Set` of strings in the constructor with the patterns and process each pattern to build a `RegexRequestMatcher` as in https://github.com/geonetwork/core-geonetwork/blob/develop/core/src/main/java/org/fao/geonet/security/web/csrf/GeonetworkCsrfSecurityRequestMatcher.java#L45 .